tonic/models/vic/grid_params.py

```diff
@@ -661,64 +1246,178 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
         bare = 1 - out_dicts['veg_dict'][var].sum(axis=0)
         bare[bare < 0.0] = 0.0
-        nveg_clases = out_dicts['veg_dict'][var].shape[0] + 1
-        shape = (nveg_clases, ) + out_dicts['veg_dict'][var].shape[1:]
+        fill_val = FILLVALUE_F
+        dtype_str = 'np.float'
+
+        # Determine the final number of veg classes, accounting for
+        # potential new bare soil class, and determine that class's idx
+        if lib_bare_idx is not None:
+            # Bare soil class already exists at lib_bare_idx
+            extra_class = 0
+        else:
+            # Bare soil class needs to be created
+            extra_class = 1
+            lib_bare_idx = out_dicts['veg_dict'][var].shape[0]
+        nveg_classes = out_dicts['veg_dict'][var].shape[0] + extra_class
+
+        # Transfer Cv info, accounting for additional bare soil areas
+        shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
         new = np.zeros(shape)
-        new[:-1, :, :] = out_dicts['veg_dict'][var]
-        new[-1, :, :] = bare
-        new /= new.sum(axis=0)
-        new[:, ymask, xmask] = FILLVALUE_F
+        if extra_class:
+            new[:-1, :, :] = out_dicts['veg_dict'][var]
+        else:
+            new[:, :, :] = out_dicts['veg_dict'][var]
+        new[lib_bare_idx, :, :] += bare
+        # Ensure that Cvs sum to 1.0
+        try:
+            new /= new.sum(axis=0)
+            new[:, ymask, xmask] = fill_val
+        except ValueError:
+            new[:, :, :] = fill_val
         out_dicts['veg_dict'][var] = new
 
-        # add dummy values for other veg vars
+        # Distribute the vegparam variables (geographically-varying)
         #   double root_depth(veg_class, root_zone, nj, ni) ;
         #   double root_fract(veg_class, root_zone, nj, ni) ;
         #   double LAI(veg_class, month, nj, ni) ;
-        for var in ['root_depth', 'root_fract', 'LAI']:
-            shape = (nveg_clases, ) + out_dicts['veg_dict'][var].shape[1:]
-            new = np.zeros(shape) + FILLVALUE_F
-            new[:-1, :, :] = out_dicts['veg_dict'][var]
-            new[-1, :, :] = 0
-            out_dicts['veg_dict'][var] = np.ma.masked_values(new, FILLVALUE_F)
+        #   double fcan(veg_class, month, nj, ni) ;
+        #   double albedo(veg_class, month, nj, ni) ;
+        fill_val = FILLVALUE_F
+        varnames = ['root_depth', 'root_fract']
+        if vegparam_lai and lai_src == 'FROM_VEGPARAM':
+            varnames.append('LAI')
+        if vegparam_fcan and fcan_src == 'FROM_VEGPARAM':
+            varnames.append('fcanopy')
+        if vegparam_albedo and alb_src == 'FROM_VEGPARAM':
+            varnames.append('albedo')
+        for var in varnames:
+            shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]
+            new = np.full(shape, fill_val)
+            if extra_class:
+                new[:-1, :, :] = out_dicts['veg_dict'][var]
+                new[-1, :, :] = bare_vegparam[var]
```

Comment:
Hmm, I'm wondering if the line `shape = (nveg_classes, ) + out_dicts['veg_dict'][var].shape[1:]` is correct...  I need to properly test that case.

But sure, I'll get rid of the trailing colons.

Question: what exactly do you mean about trailing colons not being used in numpy?  I've used them a lot and everything has worked fine.  Do you mean that trailing colons are not necessary in numpy, and that you'd like to adopt a convention in which they're not used?